core/lib/zksync_core/src/api_server/web3/namespaces/eth.rs

```diff
@@ -795,7 +800,12 @@ impl<G: L1GasPriceProvider> EthNamespace<G> {
                     )
                     .await
                     .map_err(|err| internal_error(METHOD_NAME, err))?;
-                *from_timestamp = last_timestamp.unwrap_or(*from_timestamp);
+
+                *from_timestamp = match last_timestamp {
```

Comment:
- Nit: You could've left the old logic (i.e., `unwrap_or`); it's functionally equivalent to this one and is more concise.
- Bikeshedding: maybe, it makes sense to call this field `from_timestamp_excluded` (and perhaps changing `TypedFilter` variants so it used named fields) so that it's harder to make a mistake as to the field meaning?